### PR TITLE
Integration of cluster autoscaler

### PR DESCRIPTION
**Pull request by rfranzke**
Now that we have open-sourced our work on the [cluster-autoscaler](https://github.com/gardener/autoscaler) and created the [first release](https://github.com/gardener/autoscaler/releases/tag/v0.1.0) we can finally integrate it into the Gardener.

@hardikdr @prashanth26 Please review and play with it before we merge that.

### CODE REVIEW

File: charts/seed-controlplane/charts/cluster-autoscaler/templates/deployment.yaml
Comment (prashanth26) [inline charts/seed-controlplane/charts/cluster-autoscaler/templates/deployment.yaml]: It would be better to keep this log level to --v=2 ? With --v=5, there are too many logs and we might loose older error logs if we keep them.

**Review by prashanth26 — COMMENTED** [review]
I looked through most of the changed code (except the generated and vendor changes), and it looks good to me. 

To double check, i deployed the current branch and checked the scaling up and down of nodes based on cluster load, and it looks good as well.

But I think changing log levels to `--v=2` would be better.

**Comment by prashanth26** [comment]
Resolves: https://github.com/gardener/machine-controller-manager/issues/25

**Comment by prashanth26** [comment]
Tested the PR again, looks good to me. However I would still suggest we change the logging of cluster-autoscaler to `--v=2`

**Comment by prashanth26** [comment]
LGTM. I dont think I can review/approve this PR.

**Review by prashanth26 — APPROVED** [review]
LGTM.

**Comment by rfranzke** [comment]
Waiting for @hardikdr's approval.

**Review by hardikdr — APPROVED** [review]
LGTM

**Pull request merged**
